sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -691,26 +710,22 @@ public static FileOperations getFileOperations(String spec, PipelineOptions opti
    */
   private interface FileOperations {
     /**
-     * Copy a collection of files from one location to another.
+     * Copies a collection of files from one location to another.
      *
      * <p>The number of source filenames must equal the number of destination filenames.
      *
      * @param srcFilenames the source filenames.
      * @param destFilenames the destination filenames.
      */
-     void copy(List<String> srcFilenames, List<String> destFilenames) throws IOException;
+    void copy(List<String> srcFilenames, List<String> destFilenames) throws IOException;
 
     /**
-     * Removes a directory and the files in it (but not subdirectories).
+     * Removes a collection of files or directories.
      *
-     * <p>Additionally, to partially mitigate the effects of filesystems with eventually-consistent
-     * directory matching APIs, takes a list of files that are known to exist - i.e. removes the
-     * union of the known files and files that the filesystem says exist in the directory.
-     *
-     * <p>Assumes that, if directory listing had been strongly consistent, it would have matched
-     * all of knownFiles - i.e. on a strongly consistent filesystem, knownFiles can be ignored.
+     * <p>Directories are required to be empty. Behavior of deleting non-empty directories
```

Comment:
This phrasing is wrong, because it's not "Undefined Behavior", it's just incompletely specified. Attempting to delete a non-empty directory may not succeed (but that's defined behavior). Given the current javadoc, this method may:
 * Silently return without deleting non-empty directories
 * Throw an exception

It may not
 * Delete the directory ("Directories are required to be empty")